server/src/main/java/org/elasticsearch/health/Diagnosis.java

```diff
@@ -43,11 +140,34 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         builder.field("cause", definition.cause);
         builder.field("action", definition.action);
 
-        if (affectedResources != null && affectedResources.size() > 0) {
-            builder.field("affected_resources", affectedResources);
+        if (affectedResources != null && affectedResources.length > 0) {
+            builder.startObject("affected_resources");
+            for (Resource affectedResource : affectedResources) {
+                affectedResource.toXContent(builder, params);
+            }
+            builder.endObject();
         }
 
         builder.field("help_url", definition.helpURL);
         return builder.endObject();
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        Diagnosis diagnosis = (Diagnosis) o;
+        return definition.equals(diagnosis.definition) && Arrays.equals(affectedResources, diagnosis.affectedResources);
```

Comment:
This `Diagnosis` equals/hashcode as otherwise records will be unique (unless an explicit array is passed in the constructor).
Not sure varargs are worth all this though?

Reply:
Removed them with https://github.com/elastic/elasticsearch/pull/90653/commits/279ed2affa40623067bb0f5af31a2610fe2f6d51